_includes/manuals/1.12/1.2_release_notes.md

```diff
@@ -2,6 +2,24 @@
 
 ### Headline features
 
+#### Puppet 4 support
+
+Foreman 1.12 supports Puppet 4 with the latest Puppet Agent and Puppet Server AIO packages, thanks to new features in the Foreman installer and Smart Proxy.
+
+The Smart Proxy has gained support for querying the Puppet Server API for Puppet class and parameter listings, eliminating the version-specific parser integration code and supporting AIO installations. This is activated on Puppet 4 or higher when specified in the settings file - see [Smart Proxy: Puppet](/manuals/1.12/index.html#4.3.6Puppet) for configuration information.
+
+The installer has support throughout its modules for AIO layouts, plus for automatically configuring Puppet Server 2 when installed with Puppet 4 under AIO. When installing Foreman, ensure the correct Puppet repository is configured before running the Foreman installer - see the [installation quickstart](/manuals/1.12/index.html#2.Quickstart) for instructions.
```

Comment:
I think even before installing -installer, as otherwise the old Puppet packages (from OS repo or non-PC1 PuppetLabs repo) will get installed for satisfying -installer's dependency.